### PR TITLE
Computing h value

### LINKED ISSUE #35: [TASK] Compute h
Currently, we do use h generated by the client on the preprocessing step and just PSS that to the parties. Instead, the goal of this task that we also have a way to compute h using the QAP, and it must match the h value from arkworks.

### Task checklist

1. Make sure that P, Q, and W values are correctly computed.
2. Make sure we use a valid value of t(x) (refer to arkworks repo on how that gets generated)
3. Compare the generated `h` to the one from arkworks.
4. try to get the RHS to match the LHS somehow
$$A.B - C = h(x).t(x)$$

### PR DESCRIPTION
**Pull request by shekohex**
Closes #35

### CODE REVIEW

File: groth16/src/ext_wit.rs [hunk 1]
```diff
@@ -1,12 +1,117 @@
...
+        &cd.constraint,
+        pp,
+        net,
+        CHANNEL0,
+    )
+    .await?;
+
+    let w_eval =
+        d_fft(w_coeff, false, 1, false, &cd.constraint2, pp, net, CHANNEL0)
+            .await?;
```
Comment (tbraun96) [inline groth16/src/ext_wit.rs:74]: Should be able to parallelize these operations with `tokio::try_join!`. See examples in the repo.

File: groth16/src/ext_wit.rs [hunk 2]
```diff
@@ -1,12 +1,117 @@
...
+
+    // Send the shares to the king to do the final computation
+    let received_p_shares = net.send_to_king(&p_eval, CHANNEL0).await?;
```
Comment (tbraun96) [inline groth16/src/ext_wit.rs:77]: These can also be parallelized

File: groth16/src/ext_wit.rs (hunk 1 above)
Comment (shekohex) [reply groth16/src/ext_wit.rs]: done

File: groth16/src/ext_wit.rs (hunk 2 above)
Comment (shekohex) [reply groth16/src/ext_wit.rs]: done

**Review by drewstone — APPROVED** [review]
LG

**Pull request merged**